### ISSUE TITLE
Add route for custom objects

### ISSUE THREAD

**Issue by hurradieweltgehtunter** [owncloud/marketo-client#5]
Marketo offers the ability to add custom objects to leads (http://developers.marketo.com/rest-api/lead-database/custom-objects/)
The PHP client needs to be extended to be able to use this feature.

ETA: Somewhen in the next 4 weeks, no rush.

## Create / Update
Request type: POST
URL: /rest/v1/customobjects/{objectName}.json
Body:
```
{
   "action":"{action}",
   "dedupeBy":"dedupeFields",
   "input":[
      {
         "{key1}":"{value1}",
         "{key2}":"{value2}",
         ...
      }
   ]
}
```
Parameters:
objectName: REST-API-Name of the custom object; config in Marketo
action: "createOnly", "updateOnly" or "CreateOrUpdate"
key-value: pairs of values to save; config in Marketo

## Delete
Request type: POST
URL: /rest/v1/customobjects/{objectName}/delete.json
Body:
```
{  
   "deleteBy":"dedupeFields",
   "input": {arrayOfValueObjects}
}
```
Parameters:
objectName: REST-API-Name of the custom object; config in Marketo
arrayOfValueObjects: List of objects, containing unique identifier fields (e.g. email address), like:
```
[  
  {  
    "email":"asd"
  },
]
```

**Comment by PVince81** [comment]
@IljaN can you estimate how much time needed for this ?

**Comment by IljaN** [comment]
1md

**Comment by hurradieweltgehtunter** [comment]
1md for one new api route? We should consider re-building the client if small extensions take that long.

**Comment by IljaN** [comment]
@hurradieweltgehtunter I added a 0.5md buffer in my estimate because:
- I need to add unit-tests for the new functionality
- In this extension the url  is dynamic because it is a custom object, this was not accounted for in the current design.

**Comment by PVince81** [comment]
task is blocked, waiting for review. unassigning from sprint for now

**Comment by hurradieweltgehtunter** [comment]
Yup, this is on my list. Sorry, I didn't had time yet. Will get back to you.
